provider/azure/config_test.go

```diff
@@ -44,3 +47,14 @@ func TestGetCloudConfiguration(t *testing.T) {
 		})
 	}
 }
+
+func TestOverrideConfiguration(t *testing.T) {
+	_, filename, _, _ := runtime.Caller(0)
+	configFile := path.Join(path.Dir(filename), "config_test.json")
+	cfg, err := getConfig(configFile, "subscription-override", "rg-override", "")
+	if err != nil {
+		t.Errorf("got unexpected err %v", err)
+	}
+	assert.Equal(t, cfg.SubscriptionID, "subscription-override")
+	assert.Equal(t, cfg.ResourceGroup, "rg-override")
+}
```

Comment:
Not sure if this test is useful/wanted, can also be removed

Reply:
It can help to ensures that the override still work when someone else makes a PR